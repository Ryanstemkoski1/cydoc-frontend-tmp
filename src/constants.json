{
  "medicalhistory": {
    "conditions": ["Type II Diabetes", "Myocardial Infarction", "Hypertension",  "Hypercholesteremia", "Depression", "HIV"]
  },

  "socialhistory": {
    "substanceUseContentHeader": "Substance Use",
    "secondaryFields": ["living situation", "employment", "diet", "exercise"],
    "substanceAbuseFields" : {
      "tobacco": {
        "condition": "tobacco",
        "firstField": "Packs/Day",
        "secondField": "Number of Years"
      },
      "alcohol": {
        "condition": "alcohol",
        "firstField": "Drinks/Week",
        "secondField": "What kind of drinks?"
      },
      "substanceAbuse": {
        "condition": "substance abuse",
        "firstField": "Drug Name",
        "secondField": "Mode of Delivery"
      }
    }
  }

}