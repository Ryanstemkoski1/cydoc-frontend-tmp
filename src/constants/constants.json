{
  "TAB_NAMES": [
    "HPI",
    "Patient History",
    "Review of Systems",
    "Physical Exam",
    "Plan",
    "Generate Note"
  ],
  "CONDITIONS": [
    "Type II Diabetes",
    "Myocardial Infarction",
    "Hypertension",
    "Hypercholesteremia",
    "Depression",
    "HIV"
  ],
  "MEDICAL_HISTORY": {
    "STATE": {
      "0": {
        "Condition": "Type II Diabetes",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "1": {
        "Condition": "Myocardial Infarction",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "2": {
        "Condition": "Hypertension",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "3": {
        "Condition": "Hypercholesteremia",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "4": {
        "Condition": "Depression",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      },
      "5": {
        "Condition": "HIV",
        "Yes": false,
        "No": false,
        "Onset": "",
        "Comments": ""
      }
    }
  },
  "FAMILY_HISTORY": {
    "STATE": {
      "0": {
        "Condition": "Type II Diabetes",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "1": {
        "Condition": "Myocardial Infarction",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "2": {
        "Condition": "Hypertension",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "3": {
        "Condition": "Hypercholesteremia",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "4": {
        "Condition": "Depression",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      },
      "5": {
        "Condition": "HIV",
        "Yes": false,
        "No": false,
        "Family Member": [],
        "Cause of Death": [],
        "Comments": ""
      }
    }
  },
  "SOCIAL_HISTORY": {
    "SUBSTANCE_USE_CONTENT_HEADER": "Substance Use",
    "SECONDARY_FIELDS": [
      "Living Situation",
      "Employment",
      "Diet",
      "Exercise"
    ],
    "SUBSTANCE_USE_FIELDS": {
      "Tobacco": {
        "condition": "Tobacco",
        "firstField": "Packs/Day",
        "secondField": "Number of Years",
        "thirdField": "Products Used",
        "quitYear": "Quit Year"
      },
      "Alcohol": {
        "condition": "Alcohol",
        "firstField": "Drink Type",
        "secondField": "Drink Size",
        "thirdField": "# Per Week",
        "quitYear": "Quit Year"
      },
      "Substance Abuse": {
        "condition": "Substance Abuse",
        "firstField": "Drug Name",
        "secondField": "Mode of Delivery",
        "thirdField": "# Per Week",
        "quitYear": "Quit Year"
      }
    },
    "STATE": {
      "Alcohol": {
        "Yes": false,
        "In the Past": false,
        "Never Used": false,
        "Drinks Type": "",
        "Drink Size": "",
        "# Per Week": "",
        "Quit Year": "",
        "Comments": ""
      },
      "Tobacco": {
        "Yes": false,
        "In the Past": false,
        "Never Used": false,
        "Packs/Day": "",
        "Number of Years": "",
        "Products Used": "",
        "Quit Year": "",
        "Comments": ""
      },
      "Substance Abuse": {
        "Yes": false,
        "In the Past": false,
        "Never Used": false,
        "Drug Name": "",
        "Mode of Delivery": "",
        "# Per Week": "",
        "Quit Year": "",
        "Comments": ""
      },
      "Living Situation": "",
      "Employment": "",
      "Diet": "",
      "Exercise": ""
    }
  },
  "ALLERGIES": {
    "fields": [
      "Inciting Agent",
      "Reaction",
      "Comments"
    ],
    "STATE": {
      "Inciting Agent": "",
      "Reaction": "",
      "Comments": ""
    }
  }
}