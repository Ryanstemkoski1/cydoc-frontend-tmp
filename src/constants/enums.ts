export enum YesNoResponse {
    Yes = 'YES',
    No = 'NO',
    None = '',
}

export enum YesNoMaybeResponse {
    Yes = 'YES',
    No = 'NO',
    Maybe = 'MAYBE',
    None = '',
}

export enum SubstanceUsageResponse {
    Yes = 'YES',
    InThePast = 'IN_THE_PAST',
    NeverUsed = 'NEVER_USED',
    None = '',
}

export interface LRButtonState {
    left: boolean;
    center: boolean;
    right: boolean;
}

export enum LeftRight {
    Left = 'LEFT',
    Right = 'RIGHT',
}
export enum YesNoUncertainResponse {
    Yes = 'YES',
    No = 'NO',
    Uncertain = 'UNCERTAIN',
    None = '',
}

export enum WhenResponse {
    Today = 'TODAY',
    ThisWeek = 'THIS_WEEK',
    ThisMonth = 'THIS_MONTH',
    ThisYear = 'THIS_YEAR',
    None = '',
}

export interface DropdownOption {
    key: string | number;
    value: string;
    text: string;
}
