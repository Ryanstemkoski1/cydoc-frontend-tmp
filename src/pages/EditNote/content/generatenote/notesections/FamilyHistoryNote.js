import React from 'react';

class FamilyHistoryNote extends React.Component {
    checkEmpty() {
        for (const key in this.props.familyHistory) {
            if (!(this.props.familyHistory[key].Yes === false && this.props.familyHistory[key].No === false)) {
                return false;
            }
        }
        return true;
    }

    render() {
        const familyHistory = this.props.familyHistory;

        let components = [];
        for (var condition in familyHistory) {
            let familyMembers = [];
            if (familyHistory[condition].Yes === true) {
                if (familyHistory[condition]['Family Member']) {
                    for (var member in familyHistory[condition]['Family Member']) {
                        if (familyHistory[condition]['Cause of Death'][member]) {
                            familyMembers.push(`${familyHistory[condition]['Family Member'][member]} (cause of death)`);
                        }
                        else {
                            familyMembers.push(familyHistory[condition]['Family Member'][member]);
                        }
                    }
                }
                components[condition] = {
                    condition: familyHistory[condition]['Condition'],
                    family: familyMembers,
                    comments: familyHistory[condition]['Comments']
                }
            }
        }

        if (this.checkEmpty()) {
            return (
                <div>No family history reported.</div>
            );
        }

        return (
            <ul>
                {Object.keys(components).map(key => (
                    <li>
                        <b>{components[key].condition}: </b>
                        {components[key].family.length > 0 ? `${components[key].family.join(', ')}. ` : null} 
                        {components[key].comments}
                    </li>
                ))}
            </ul>
        )
    }
}

export default FamilyHistoryNote;