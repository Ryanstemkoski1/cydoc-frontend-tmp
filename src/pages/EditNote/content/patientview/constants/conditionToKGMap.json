{
    "ADHD": "ADHD",
    "Depression": "DEPRESSION",
    "Genital Symptoms": "SEXUALINFECTION",
    "High Blood Pressure": "HYPERTENSION",
    "High Cholesterol": "HYPERLIPIDEMIA",
    "High Triglycerides": "HYPERLIPIDEMIA",
    "Urinary Symptoms": "URINARY_SYMPTOMS"
}
