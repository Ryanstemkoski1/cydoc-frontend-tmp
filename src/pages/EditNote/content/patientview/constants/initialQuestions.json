{
    "order": {
        "1": "1",
        "2": "2",
        "3": "3",
        "4": "4",
        "5": "5",
        "6": "6",
        "7": "7"
    },
    "graph": {
        "1": ["2", "3", "4", "8", "9"],
        "2": [],
        "3": [],
        "4": [],
        "5": ["6", "7"],
        "6": [],
        "7": []
    },
    "nodes" : {
            "1": {
                "uid": "1",
                "text": "",
                "responseType": "",
                "category": "",
                "doctorView": ""
            },
            "2": {
                "uid": "2",
                "text": "Are you here for an annual physical exam?",
                "responseType": "YES-NO",
                "category": "ANNUALVISIT",
                "doctorView": "Annual Physical Exam"
            },
            "3": {
                "uid": "3",
                "text": "Are you here to follow up any previously diagnosed medical conditions?",
                "responseType": "YES-NO",
                "category": "",
                "doctorView": ""
            },
            "4": {
                "uid": "4",
                "text": "Do you have any new health concerns to discuss with the clinician today?",
                "responseType": "YES-NO",
                "category": "",
                "doctorView": ""
            },
            "5": {
                "uid": "5",
                "text": "",
                "responseType": "",
                "category": "",
                "doctorView": ""
            },
            "6": {
                "uid": "6",
                "text": "Please select the most important conditions or symptoms that you would like to discuss today, up to a maximum of 3. CLICK[ADHD, Depression, Diabetes, Genital Symptoms/STI, Hypertension, Hyperlipidemia, Urinary Symptoms/UTI]",
                "responseType": "SELECTONE",
                "category": "",
                "doctorView": ""
            },
            "7": {
                "uid": "7",
                "text": "If youâ€™d like to discuss an additional condition or symptom not listed above, please search for it here:",
                "responseType": "SEARCH",
                "category": "",
                "doctorView": ""
            },
            "8": {
                "uid": "8",
                "text": "Please confirm the date of your appointment.",
                "responseType": "TIME3DAYS",
                "category": "",
                "doctorView": ""
            },
            "9": {
                "uid": "9",
                "text": "What is the last name of the clinician you are seeing today? (optional)",
                "responseType": "SHORT-TEXT",
                "category": "",
                "doctorView": ""
            }
        }
}
