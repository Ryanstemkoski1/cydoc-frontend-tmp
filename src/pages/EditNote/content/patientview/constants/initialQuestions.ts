import { ResponseTypes } from 'constants/hpiEnums';
import { InitialQuestionsState } from 'redux/reducers/userViewReducer';

const initialQuestions: InitialQuestionsState = {
    order: {
        '1': '1',
        '2': '2',
        '3': '3',
        '4': '4',
        '5': '5',
        '6': '6',
        '7': '7',
    },
    graph: {
        '1': ['8', '9'],
        '2': [],
        '3': [],
        '4': [],
        '5': ['6', '7'],
        '6': [],
        '7': [],
    },
    nodes: {
        '1': {
            uid: '1',
            text: '',
            responseType: ResponseTypes.NULL,
            category: '',
            doctorView: '',
        },
        '2': {
            uid: '2',
            text: 'Are you here for an annual physical exam?',
            responseType: ResponseTypes.YES_NO,
            category: 'ANNUALVISIT',
            doctorView: 'Annual Physical Exam',
        },
        '3': {
            uid: '3',
            text: 'Are you here to follow up any previously diagnosed medical conditions?',
            responseType: ResponseTypes.YES_NO,
            category: '',
            doctorView: '',
        },
        '4': {
            uid: '4',
            text: 'Do you have any new health concerns to discuss with the clinician today?',
            responseType: ResponseTypes.YES_NO,
            category: '',
            doctorView: '',
        },
        '5': {
            uid: '5',
            text: '',
            responseType: ResponseTypes.NULL,
            category: '',
            doctorView: '',
        },
        '6': {
            uid: '6',
            text: 'Please select the most important conditions or symptoms that you would like to discuss today, up to a maximum of 3.',
            responseType: ResponseTypes.SELECTONE,
            category: '',
            doctorView: '',
        },
        '7': {
            uid: '7',
            text: 'List any other conditions or symptoms you would like to discuss with the clinician today. Please be accurate as this information will be used to organize your visit.',
            responseType: ResponseTypes.LIST_TEXT,
            category: '',
            doctorView: '',
        },
        '8': {
            uid: '8',
            text: 'Please confirm the date of your appointment.',
            responseType: ResponseTypes.TIME3DAYS,
            category: '',
            doctorView: '',
        },
        '9': {
            uid: '9',
            text: 'What is the last name of the clinician you are seeing today? (optional)',
            responseType: ResponseTypes.SHORT_TEXT,
            category: '',
            doctorView: '',
        },
    },
};

export default initialQuestions;
