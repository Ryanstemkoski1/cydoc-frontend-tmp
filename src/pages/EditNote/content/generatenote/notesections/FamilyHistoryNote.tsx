import { YesNoResponse } from 'constants/enums';
import React, { Component } from 'react';
import { FamilyHistoryState } from 'redux/reducers/familyHistoryReducer';
import { Table } from 'semantic-ui-react';

interface FamilyHistoryProps {
    familyHistory: FamilyHistoryState;
    isRich: boolean;
}

export class FamilyHistoryNote extends Component<FamilyHistoryProps> {
    checkEmpty = () => {
        for (const id in this.props.familyHistory) {
            if (
                this.props.familyHistory[id].hasAfflictedFamilyMember !==
                YesNoResponse.None
            ) {
                return false;
            }
        }
        return true;
    };

    render() {
        const { familyHistory } = this.props;

        const components: {
            [conditionIndex: string]: {
                [condition: string]: any;
            };
        } = {};
        const noHistory: string[] = [];
        for (const conditionIndex in familyHistory) {
            const familyMembersArray: string[] = [];
            const familyCondition = familyHistory[conditionIndex];
            // if condition has afflicted family members
            if (
                familyCondition.hasAfflictedFamilyMember === YesNoResponse.Yes
            ) {
                // if family members are listed
                if (familyCondition.familyMembers) {
                    // loop through each family member
                    for (const index in familyCondition.familyMembers) {
                        const familyMember =
                            familyCondition.familyMembers[index];
                        let status = '';

                        if (familyMember.causeOfDeath === YesNoResponse.Yes) {
                            status = `(died of ${familyCondition.condition})`;
                        } else if (
                            familyMember.causeOfDeath === YesNoResponse.No &&
                            familyMember.living === YesNoResponse.Yes
                        ) {
                            status = '(still living)';
                        } else if (
                            familyMember.causeOfDeath === YesNoResponse.No &&
                            familyMember.living === YesNoResponse.No
                        ) {
                            status = '(not the cause of death)';
                        }
                        familyMembersArray.push(
                            `${familyMember.member} ${status}${
                                familyMember.comments ? ', ' : ''
                            }${familyMember.comments}`
                        );
                    }
                }

                components[conditionIndex] = {
                    condition: familyCondition.condition,
                    family: familyMembersArray,
                };
            } else if (
                familyCondition.hasAfflictedFamilyMember === YesNoResponse.No
            ) {
                if (familyCondition.condition !== '') {
                    noHistory.push(familyCondition.condition);
                }
            }
        }

        if (this.checkEmpty()) {
            return <div />;
        } else if (!this.props.isRich) {
            return (
                <ul>
                    {Object.keys(components).map((key, index) => {
                        const conditionText =
                            components[key].family.length > 0 &&
                            components[key].family[0] !== ' '
                                ? `${components[key].family.join(', ')}. `
                                : '';
                        return (
                            <li key={index}>
                                <b>
                                    {components[key].condition}
                                    {conditionText.length > 0 ? ':' : ''}{' '}
                                </b>
                                {conditionText}
                            </li>
                        );
                    })}
                    {noHistory && noHistory.length >= 1 ? (
                        <li>
                            No family history of{' '}
                            {noHistory.length > 1
                                ? noHistory.slice(0, -1).join(', ') +
                                  ', or ' +
                                  noHistory.slice(-1)
                                : noHistory[0]}
                            .
                        </li>
                    ) : null}
                </ul>
            );
        } else {
            return (
                <Table>
                    <Table.Header>
                        <Table.Row>
                            <Table.HeaderCell>Condition</Table.HeaderCell>
                            <Table.HeaderCell>Family History</Table.HeaderCell>
                        </Table.Row>
                    </Table.Header>
                    <Table.Body>
                        {Object.keys(components).map((key, index) => {
                            const conditionText =
                                components[key].family.length > 0 &&
                                components[key].family[0] !== ' '
                                    ? `${components[key].family.join('; ')}. `
                                    : '';
                            return (
                                <Table.Row key={index}>
                                    <Table.Cell>
                                        {<b>{components[key].condition}</b>}
                                    </Table.Cell>
                                    <Table.Cell>{conditionText}</Table.Cell>
                                </Table.Row>
                            );
                        })}
                        {noHistory ? (
                            <Table.Row>
                                <Table.Cell>No family history of</Table.Cell>
                                <Table.Cell>
                                    {noHistory.length > 1
                                        ? noHistory.slice(0, -1).join(', ') +
                                          ', or ' +
                                          noHistory.slice(-1)
                                        : noHistory[0]}
                                    .
                                </Table.Cell>
                            </Table.Row>
                        ) : null}
                    </Table.Body>
                </Table>
            );
        }
    }
}

export default FamilyHistoryNote;
