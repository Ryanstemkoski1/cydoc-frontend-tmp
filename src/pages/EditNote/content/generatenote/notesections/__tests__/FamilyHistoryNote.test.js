import React from 'react';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import { FamilyHistoryNote } from '../FamilyHistoryNote';
import { YesNoResponse } from 'constants/enums';
import { FamilyOption } from 'constants/familyHistoryRelations';

Enzyme.configure({ adapter: new EnzymeAdapter() });

// Empty indicated by all hasAfflictedFamilyMember == YesNoReponse.None
const emptyFH = {
    0: {
        condition: 'Type II Diabetes',
        hasAfflictedFamilyMember: YesNoResponse.None,
        familyMembers: {},
    },
};

const mountWithProps = (familyHistory = emptyFH) => {
    return mount(<FamilyHistoryNote familyHistory={familyHistory} />);
};

describe('Family History Note', () => {
    it('renders without crashing', () => {
        const wrapper = mountWithProps();
        expect(wrapper).toBeTruthy();
    });
    it('matches snapshot', () => {
        const wrapper = mountWithProps();
        expect(wrapper.html()).toMatchSnapshot();
    });
    it('renders correctly when completely empty', () => {
        const wrapper = mountWithProps();
        expect(wrapper.text()).toContain('');
    });
    it('renders only non-empty entries', () => {
        const wrapper = mountWithProps({
            0: {
                condition: 'foo',
                hasAfflictedFamilyMember: YesNoResponse.Yes,
                familyMembers: {
                    0: {
                        member: FamilyOption.Mother,
                        causeOfDeath: YesNoResponse.Yes,
                        living: YesNoResponse.None,
                        comments: 'comment',
                    },
                    1: {
                        member: FamilyOption.Son,
                        causeOfDeath: YesNoResponse.No,
                        living: YesNoResponse.Yes,
                        comments: '',
                    },
                },
            },
            1: {
                // Should not see because hasAfflicatedFamilyMember == YesNoResponse.No
                condition: 'foo',
                hasAfflictedFamilyMember: YesNoResponse.No,
                familyMembers: {
                    0: {
                        member: FamilyOption.Mother,
                        causeOfDeath: YesNoResponse.Yes,
                        living: YesNoResponse.None,
                        comments: 'comment',
                    },
                },
            },
            2: {
                condition: 'bar',
                hasAfflictedFamilyMember: YesNoResponse.Yes,
                familyMembers: {
                    0: {
                        member: FamilyOption.MaternalAunt,
                        causeOfDeath: YesNoResponse.No,
                        living: YesNoResponse.No,
                        comments: 'comment',
                    },
                },
            },
            3: {
                condition: 'lonely',
                hasAfflictedFamilyMember: YesNoResponse.Yes,
                familyMembers: {},
            },
        });
        const expected = [
            'foo: mother (died of foo), comment, son (still living). ',
            'bar: maternal aunt (not the cause of death), comment. ',
            'lonely ',
            'No family history of foo.',
        ];
        const items = wrapper.find('li');
        expect(items).toHaveLength(4);
        items.forEach((node, i) => expect(node.text()).toEqual(expected[i]));
    });
});
