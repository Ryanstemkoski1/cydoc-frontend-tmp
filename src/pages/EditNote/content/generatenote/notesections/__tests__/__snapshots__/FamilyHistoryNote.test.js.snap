// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Family History Note matches snapshot 1`] = `"<div></div>"`;
